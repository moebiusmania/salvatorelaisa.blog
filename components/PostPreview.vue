<script setup lang="ts">
import type { ParsedContent } from "@nuxt/content";
import { PINNED_POST_DESCRIPTION } from "~/utils/config";

defineProps<{
  post: ParsedContent;
}>();
</script>

<template>
  <li :class="{ 'pinned': post.pinned }">
    <span v-if="post.pinned" class="badge" :title="PINNED_POST_DESCRIPTION">in evidenza</span>
    <time>
      <span>
        {{
          new Date(post.date).toLocaleDateString("it-it", {
            year: "numeric",
            month: "long",
            day: "numeric",
          })
        }}
      </span>
    </time>
    <div>
      <NuxtLink :href="`/post${post._path}`">
        <h2>
          {{ post.title }}
        </h2>
      </NuxtLink>
      <div>
        <NuxtLink v-for="tag in post.tags" :href="`/tags/${tag}`">
          {{ tag }}
        </NuxtLink>
      </div>
      <p>
        {{ post.summary }}
      </p>
      <NuxtLink :href="`/post${post._path}`">
        Continua a leggere ({{
          post.readingTime.text.replace("min read", "minuti")
        }})
        <svg viewBox="0 0 24 24" stroke="currentColor" stroke-width="2" fill="none" stroke-linecap="round"
          stroke-linejoin="round">
          <path d="M5 12h14"></path>
          <path d="M12 5l7 7-7 7"></path>
        </svg>
      </NuxtLink>
    </div>
  </li>
</template>

<style scoped>
li {
  display: flex;
  padding: var(--sp-4) 0;
  flex-wrap: wrap;
  position: relative;

  &.pinned {
    margin: var(--sp-4) 0;
    border: 1px solid var(--primary);
    border-top: 10px solid var(--primary);
    border-bottom: 10px solid var(--primary);
    padding: var(--sp-2);
    /* box-shadow: 0 0 20px 0 var(--primary); */

    @media (min-width: 768px) {
      border: 1px solid var(--primary);
      border-left: 10px solid var(--primary);
      padding-left: var(--sp-1);
    }

    & .badge {
      position: absolute;
      top: -35px;
      left: -1px;
      background-color: var(--primary);
      color: var(--white);
      padding: var(--sp-1);
      cursor: help;
      text-transform: uppercase;

      @media (min-width: 768px) {
        top: unset;
        bottom: 0;
      }
    }

    &>time {
      @media (min-width: 768px) {
        width: 14rem;
      }
    }
  }

  @media (min-width: 768px) {
    flex-wrap: nowrap;
  }

  &>time {
    display: flex;
    margin-bottom: var(--sp-3);
    flex-direction: column;

    @media (min-width: 768px) {
      margin-bottom: 0;
      width: 16rem;
    }

    & span {
      margin-top: var(--sp-05);
      color: var(--secondary)
    }
  }

  &>div:last-child {
    width: 100%;

    @media (min-width: 768px) {
      flex-grow: 1;
    }

    & h2 {
      margin-top: 0;
      margin-bottom: var(--sp-1);
      color: var(--text-base-content)
    }

    &>div {
      display: flex;
      flex-wrap: wrap;

      &>a {
        margin-right: var(--sp-2);
        font-size: 0.875rem;
        line-height: 1.25rem;
        font-weight: 500;
        text-transform: uppercase;
        color: var(--primary);

        &:hover {
          text-decoration: underline;
        }
      }
    }

    & p {
      margin-top: var(--sp-3);
      line-height: 1.625;
    }

    &>a:last-child {
      display: inline-flex;
      margin-top: var(--sp-2);
      align-items: center;
      color: var(--primary);

      & svg {
        margin-left: var(--sp-1);
        width: 1rem;
        height: 1rem;
      }
    }
  }
}
</style>