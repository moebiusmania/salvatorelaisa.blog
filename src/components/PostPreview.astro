---
export interface Props {
  post: Record<string, any>
}

const { post } = Astro.props as Props
---

<li class="py-8 flex flex-wrap md:flex-nowrap">
  <div class="md:w-64 md:mb-0 mb-6 flex-shrink-0 flex flex-col">
    <span class="mt-1 text-secondary">
      {
        new Date(post.frontmatter.date).toLocaleDateString('it-it', {
          year: 'numeric',
          month: 'long',
          day: 'numeric',
        })
      }
    </span>
  </div>
  <div class="md:flex-grow">
    <a href={post.url}>
      <h2 class="text-2xl font-medium text-base-content title-font mb-2">
        {post.frontmatter.title}
      </h2>
    </a>
    <div class="flex flew-wrap">
      {
        post.frontmatter.tags.map((tag: string) => (
          <a
            class="mr-3 text-sm font-medium text-primary uppercase hover:text-primary-focus"
            href="/tags/film"
            href={`/tags/${tag}`}
          >
            {tag}
          </a>
        ))
      }
    </div>
    <p class="leading-relaxed mt-6 dark:text-base-content">{post.frontmatter.summary}</p>
    <a class="text-primary hover:text-primary-focus inline-flex items-center mt-4" href={post.url}>
      Continua a leggere
      <svg
        class="w-4 h-4 ml-2"
        viewBox="0 0 24 24"
        stroke="currentColor"
        stroke-width="2"
        fill="none"
        stroke-linecap="round"
        stroke-linejoin="round"
      >
        <path d="M5 12h14"></path>
        <path d="M12 5l7 7-7 7"></path>
      </svg>
    </a>
  </div>
</li>
