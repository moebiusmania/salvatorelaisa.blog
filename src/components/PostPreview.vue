<script setup lang="ts">
import { reading } from '../utils'

defineProps<{
  post: Record<string, any>
}>()
</script>

<template>
  <li class="py-8 flex flex-wrap md:flex-nowrap">
    <div class="md:w-64 md:mb-0 mb-6 flex-shrink-0 flex flex-col">
      <span class="mt-1 text-secondary">
        {{
          new Date(post.frontmatter.date).toLocaleDateString('it-it', {
            year: 'numeric',
            month: 'long',
            day: 'numeric',
          })
        }}
      </span>
    </div>
    <div class="md:flex-grow">
      <a :href="post.url">
        <h2 class="text-2xl font-medium text-base-content title-font mb-2">
          {{ post.frontmatter.title }}
        </h2>
      </a>
      <div class="flex flew-wrap">
        <a
          v-for="tag in post.frontmatter.tags"
          class="mr-3 text-sm font-medium text-primary uppercase hover:text-primary-focus"
          :href="`/tags/${tag}`"
        >
          {{ tag }}
        </a>
      </div>
      <p class="leading-relaxed mt-6 dark:text-base-content">{{ post.frontmatter.summary }}</p>
      <a
        class="text-primary hover:text-primary-focus inline-flex items-center mt-4"
        :href="post.url"
      >
        Continua a leggere ({{ reading(post.rawContent) }})
        <svg
          class="w-4 h-4 ml-2"
          viewBox="0 0 24 24"
          stroke="currentColor"
          stroke-width="2"
          fill="none"
          stroke-linecap="round"
          stroke-linejoin="round"
        >
          <path d="M5 12h14"></path>
          <path d="M12 5l7 7-7 7"></path>
        </svg>
      </a>
    </div>
  </li>
</template>
