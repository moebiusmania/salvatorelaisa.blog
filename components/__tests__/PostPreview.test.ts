import { describe, it, expect } from "vitest";
import { mount } from "@vue/test-utils";
import PostPreview from "../PostPreview.vue";

describe("PostPreview", () => {
  const mockPost = {
    _path: "/test-post",
    title: "Test Post Title",
    date: "2024-03-20T10:00:00.000Z",
    tags: ["test", "vue"],
    summary: "This is a test post summary",
    readingTime: {
      text: "5 min read",
    },
  };

  // Snapshot test
  it("renders correctly", () => {
    const wrapper = mount(PostPreview, {
      props: {
        post: {
          ...mockPost,
          body: {
            type: "root",
            children: [],
          },
          _id: "test-id",
        },
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
