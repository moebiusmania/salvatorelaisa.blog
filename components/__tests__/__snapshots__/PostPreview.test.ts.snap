// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PostPreview > renders correctly 1`] = `
"<li data-v-ef65044c="">
  <div data-v-ef65044c=""><span data-v-ef65044c="">20 marzo 2024</span></div>
  <div data-v-ef65044c="">
    <routerlink data-v-ef65044c="" to="/post/test-post"></routerlink>
    <div data-v-ef65044c="">
      <routerlink data-v-ef65044c="" to="/tags/test"></routerlink>
      <routerlink data-v-ef65044c="" to="/tags/vue"></routerlink>
    </div>
    <p data-v-ef65044c="">This is a test post summary</p>
    <routerlink data-v-ef65044c="" to="/post/test-post"></routerlink>
  </div>
</li>"
`;
